### ISSUE TITLE
Multiple accounts on the same instance

### ISSUE THREAD

**Issue by nolanlawson** [nolanlawson/pinafore#39]
This _might_ be possible, but would be really tricky given how I implemented data sharding on a per-instance basis.

**Comment by nightpool** [comment]
this is a big plus 1 for me, it's the only reason I really have to use a non-stock pwa

**Comment by nightpool** [comment]
use "username@domain" as the sharding key instead of just "domain"?

**Comment by nolanlawson** [comment]
Yes, it would require an IndexedDB database migration, but it's totally doable. (And the DB migration is not a huge deal, since it's really just a lite cache and doesn't store any critical data anyway.)

**Comment by eloquence** [comment]
Also just writing to express that this would be very valuable, since the native web interface doesn't support it. As someone who maintains a bot + a couple regular accounts, I have to jockey browser profiles to make this work right now. In any event, thanks for making such an elegant alternative UI!

**Comment by Fidgetcetera** [comment]
Chiming in to say that I'd also prefer this a lot, I'm using 3 accounts on one instance and pinafore looks otherwise absolutely great to use.

**Comment by gergelypolonkai** [comment]
Another “me too!” here; I just created an account for my new project where I toot development progress, and it happens to be on the same instance as my primary one. Right now I use a private tab for this (and mobile apps, obviously.) Being able to do this in my browser would be awesome.

**Comment by desvox** [comment]
yet another me too! I would love to switch between my public account and private account from the same pinafore session

**Comment by bnjbvr** [comment]
Another +1 from me!

Assuming all the data living in indexeddb is settings + replicated toots and toot metadata, could it be possible to keep the settings, change the key only for toots, and trash the replicated data? It would probably make an easier migration. Or is it just the fact that it needs a migration that makes the whole process complicated, and then migrating the data itself wouldn't be so complicated?

**Comment by Hund** [comment]
I would love to see this feature implemented! :)

**Comment by PhieF** [comment]
+A
